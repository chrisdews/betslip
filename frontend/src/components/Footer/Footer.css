.footer {
    background-color: aqua;
    display: flex;
    justify-content: space-between;
    align-items: right;
    padding: 0em 1em 0em 1em;
}

.betButton {
    margin-left: auto;
}