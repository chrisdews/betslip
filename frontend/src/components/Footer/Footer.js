import React from "react";
import PropTypes from "prop-types";
import Button from "../Button";
import styled from "styled-components";
function Footer({ betButtonDisabled, setBetPlaced }) {
  const placeBetClickHandler = () => {
    setBetPlaced(true);
  };
  return (
    <FooterStyle>
      <BetButton>
        <Button
          clickHandler={placeBetClickHandler}
          buttonDisabled={betButtonDisabled}
        >
          Bet Now
        </Button>
      </BetButton>
    </FooterStyle>
  );
}

Footer.propTypes = {
  betButtonDisabled: PropTypes.bool,
  setBetPlaced: PropTypes.func,
};

export default Footer;

const FooterStyle = styled.div`
  background-color: aqua;
  display: flex;
  justify-content: space-between;
  align-items: right;
  padding: 0em 1em 0em 1em;
`;

const BetButton = styled.span`
  margin-left: auto;
`;
