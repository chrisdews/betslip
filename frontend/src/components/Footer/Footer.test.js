import React from 'react';
import renderer from 'react-test-renderer';
import Footer from './Footer';
import { render, screen } from '@testing-library/react'

describe ('Footer', ()=>{
    it('renders correctly - snapshot', () => {
        const tree = renderer
          .create(<Footer><button>Bet Now</button></Footer>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

    test('renders bet now button', () => {
        render(<Footer><button>Bet Now</button></Footer>)
        expect(screen.getByText('Bet Now'))
    })
})