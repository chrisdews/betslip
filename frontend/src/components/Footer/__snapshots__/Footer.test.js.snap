// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly - snapshot 1`] = `
<div
  className="sc-gtsrHT kgtcCO"
>
  <span
    className="sc-dlnjwi kcWHpo"
  >
    <button
      className="sc-bdnxRM cLrBHU"
      onClick={[Function]}
    >
      Bet Now
    </button>
  </span>
</div>
`;
